src/renderer/components/badge/badge.tsx

```diff
@@ -3,21 +3,43 @@ import "./badge.scss"
 import React from "react";
 import { cssNames } from "../../utils/cssNames";
 import { TooltipDecoratorProps, withTooltip } from "../tooltip";
+import { observer } from "mobx-react";
+import { autobind } from "../../utils";
+import { observable } from "mobx";
 
 export interface BadgeProps extends React.HTMLAttributes<any>, TooltipDecoratorProps {
   small?: boolean;
   label?: React.ReactNode;
+  isExpanded?: boolean; // by default: false (minimized)
 }
 
 @withTooltip
+@observer
 export class Badge extends React.Component<BadgeProps> {
+  @observable isExpanded = false;
+
+  @autobind()
+  onMouseRelease() {
+    const textSelected = document.getSelection().toString().trim() !== "";
+    if (textSelected) return;
```

Comment:
If you highlight and then stop highlighting text and then release, it still closes. This is pretty weird.

Reply:
Nope, can't reproduce. Note: it's bound to `onMouseUp`

Reply:
![Screen Recording 2020-11-16 at 1 29 56 PM](https://user-images.githubusercontent.com/8225332/99293136-35ac3c80-2810-11eb-8b16-71cd0583454a.gif)

Reply:
Yes, in your example actual selection won't happen, so why not? Very rare use-case imho (to return mouse to the start and just like "oh, i have changed by mind"?)
Could be catched easily with `document.onselectionchange`.

Reply:
Maybe, but I am talking about intent. The user probably would find this jarring since he didn't intend to close the badge.